### PR TITLE
Allow grant creators to "speed up" contract deployment txn

### PR DESCRIPTION
**Pull request by danlipert**
##### Description

When a new grant is created, if the grant creator "speeds up" the transaction in metamask due to low gas prices, the new transaction is not monitored for the contract address and grant creation fails. This PR fixes this issue.

##### Checklist

- [x] Read and conforms to the [contributor guidelines](https://docs.gitcoin.co/mk_contributors/)
- [x] Read and conforms to the [style guidelines](https://docs.gitcoin.co/mk_styleguide/)
- [x] Linter status: 100% pass
- [x] Added relevant tests (N/A)
- [x] Tests pass and code coverage has not decreased (N/A)
- [x] Changes don't break existing behavior
- [x] Commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)
- [x] [Code documentation](https://docs.gitcoin.co/mk_contributors/#docstrings)
- [ ] Completed: Code review by core team and any requested changes

##### Affected core subsystem(s)

Grants

##### Refers/Fixes

Fixes: #3524 

##### Testing and Sign-off

See video in comments for example of behavior

###### Contributor

- [x] Read and followed the [Contributor Guidelines](https://docs.gitcoin.co/mk_contributors/)
- [x] Tested all changes **locally**
- [x] Verified existing functionality
- [x] Ran `make test` and everything passed!

###### Reviewer

- [ ] Affirm contributor guidelines have been followed and requested changes made
- [ ] CI tests and linting pass
- [ ] No conflicts (migrations, files, etc)
- [ ] Regression tested against staging or local deployment

###### Funder

- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

### CODE REVIEW

**Comment by danlipert** [comment]
Video of initial "working" version: https://www.youtube.com/watch?v=uavTvbru4PY

**Comment by danlipert** [comment]
Video with console logging shown: https://www.youtube.com/watch?v=DPEOzM5sJ3k

File: app/assets/v2/js/grants/new.js [hunk 1]
```diff
@@ -98,65 +125,75 @@ const init = () => {
...
+                      setTimeout(() => {
+                        checkForContractCreation(transactionHash);
+                      }, 1000);
+                    } else if (block && block.transactions) {
+                      checkedBlocks.push(blockToCheck);
+                      blockToCheck = blockToCheck + 1;
+
+                      let didFindTransaction = false;
+
+                      for (var i = 0; i < block.transactions.length; i += 1) {
```
Comment (thelostone-mc) [inline app/assets/v2/js/grants/new.js]: use `let` ?

File: app/assets/v2/js/grants/new.js [hunk 2]
```diff
@@ -98,65 +125,75 @@ const init = () => {
...
+                        if (block.transactions[i].from == accounts[0]) {
+                          didFindTransaction = true;
+                          web3.eth.getTransactionReceipt(block.transactions[i].hash, function(error, result) {
```
Comment (thelostone-mc) [inline app/assets/v2/js/grants/new.js]: use arrow function to keep it uniform ?

**Comment by danlipert** [comment]
> Looks fine locally !
> Just minor comments + conflict resolution and we should we good

@thelostone-mc Sweet - updated! Please take a look and let me know what you think

**Pull request merged**